lib/crypto/random.js

```diff
@@ -4,4 +4,32 @@
 
 'use strict'
 
-module.exports = require('../promise').promisify(require('crypto').randomBytes)
+const randomBytes = require('../promise').promisify(require('crypto').randomBytes)
+
+function random(bytes) {
+  if (Array.isArray(bytes)) {
```

Comment:
I second this, switching behaviour based on the type of an argument has been a footgun in the past.  I wonder if this should just be two separate functions, one for a single string and one for an array.  Or, have the caller call the function inside a promisified map()?